### ISSUE TITLE
TypeError: can't dup NilClass on v5.0.2+

### ISSUE THREAD

**Issue by Xosmond** [redis-store/redis-activesupport#99]
Command:
`bundle exec rake assets:precompile --trace`
Error:
```
** Invoke assets:precompile (first_time)
** Invoke assets:environment (first_time)
** Execute assets:environment
** Invoke environment (first_time)
** Execute environment
rake aborted!
TypeError: can't dup NilClass
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/redis-activesupport-5.0.4/lib/active_support/cache/redis_store.rb:46:in `dup'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/redis-activesupport-5.0.4/lib/active_support/cache/redis_store.rb:46:in `map'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/redis-activesupport-5.0.4/lib/active_support/cache/redis_store.rb:46:in `initialize'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/activesupport-5.1.4/lib/active_support/cache.rb:57:in `new'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/activesupport-5.1.4/lib/active_support/cache.rb:57:in `lookup_store'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/application/bootstrap.rb:65:in `block in <module:Bootstrap>'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/initializable.rb:30:in `instance_exec'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/initializable.rb:30:in `run'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/initializable.rb:59:in `block in run_initializers'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:228:in `block in tsort_each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:431:in `each_strongly_connected_component_from'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:349:in `block in each_strongly_connected_component'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:347:in `each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:347:in `call'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:347:in `each_strongly_connected_component'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:226:in `tsort_each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/tsort.rb:205:in `tsort_each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/initializable.rb:58:in `run_initializers'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/application.rb:353:in `initialize!'
/home/user/app/config/environment.rb:5:in `<top (required)>'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `block in require'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:258:in `load_dependency'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/activesupport-5.1.4/lib/active_support/dependencies.rb:292:in `require'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/application.rb:329:in `require_environment!'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/railties-5.1.4/lib/rails/application.rb:445:in `block in run_tasks_blocks'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `block in execute'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `execute'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:195:in `block in invoke_with_call_chain'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:188:in `invoke_with_call_chain'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:181:in `invoke'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/sprockets-rails-3.2.1/lib/sprockets/rails/task.rb:62:in `block (2 levels) in define'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `block in execute'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:251:in `execute'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:195:in `block in invoke_with_call_chain'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:188:in `invoke_with_call_chain'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:217:in `block in invoke_prerequisites'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:215:in `each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:215:in `invoke_prerequisites'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:194:in `block in invoke_with_call_chain'
/home/user/.rbenv/versions/2.3.5/lib/ruby/2.3.0/monitor.rb:214:in `mon_synchronize'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:188:in `invoke_with_call_chain'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/task.rb:181:in `invoke'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:160:in `invoke_task'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:116:in `each'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:116:in `block in top_level'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:125:in `run_with_threads'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:110:in `top_level'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:83:in `block in run'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:186:in `standard_exception_handling'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/lib/rake/application.rb:80:in `run'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/rake-12.3.0/exe/rake:27:in `<top (required)>'
/home/user/.rbenv/versions/2.3.5/bin/rake:22:in `load'
/home/user/.rbenv/versions/2.3.5/bin/rake:22:in `<top (required)>'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
/home/user/.rbenv/versions/2.3.5/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
/home/user/.rbenv/versions/2.3.5/bin/bundle:22:in `load'
/home/user/.rbenv/versions/2.3.5/bin/bundle:22:in `<main>'
Tasks: TOP => environment
```

I had to take it back to version 5.0.1 and no error is raised.

**Comment by tubbo** [comment]
@Xosmond It seems the `addresses` splat argument passed in was `nil`, so that might be a reason. I'm not sure why reverting to 5.0.1 would fix the issue though, there have been no changes to that particular line of code between 5.0.0 and 5.0.4.

**Comment by Xosmond** [comment]
Hello @tubbo, I have checked and this is the commit https://github.com/redis-store/redis-activesupport/commit/cb481440ca44cf8d5f5c22c3a9ad56fb475f7ab0, that added the line that is generating the error.

**Comment by Xosmond** [comment]
Hello @tubbo, I have checked and this is the commit https://github.com/redis-store/redis-activesupport/commit/cb481440ca44cf8d5f5c22c3a9ad56fb475f7ab0, that added the line that is generating the error.

This commit is added to version 5.0.2

**Comment by tubbo** [comment]
ahh I see what you mean. We can probably make that safer. Thanks for letting me know!
  

  
  

  
>

**Comment by tubbo** [comment]
@Xosmond how are you setting up your rails cache? can i see your `config.cache_store` value (with secrets redacted of course)?

**Comment by tubbo** [comment]
@Xosmond I just merged #91 into master. you should be able to update using the following line in Gemfile:

```ruby
gem 'redis-activesupport', github: 'redis-store/redis-activesupport'
```
